provision/pool/constraint.go

```diff
@@ -263,3 +266,33 @@ func ToConstraintType(value string) (poolConstraintType, error) {
 	}
 	return poolConstraintType(value), nil
 }
+
+func CheckServiceAllowed(ctx context.Context, pool, service string) error {
+	p, err := GetPoolByName(ctx, pool)
+	if err != nil {
+		return err
+	}
+
+	poolAllowedServices, err := p.GetServices()
+	if err != nil {
+		return err
+	}
+
+	if !sliceContains(poolAllowedServices, service) {
+		return &tsuruErrors.HTTP{
+			Code:    http.StatusBadRequest,
+			Message: fmt.Sprintf("Service %q is not available in pool %q", service, pool),
+		}
```

Comment:
WDYT returning an error message with the available services as well? Pretty much like we do in routers validation (see [`ValidateRouters`](https://github.com/tsuru/tsuru/blob/main/provision/pool/pool.go#L247))